src/Middleware/RequestDecompression/src/RequestDecompressionProvider.cs

```diff
@@ -0,0 +1,144 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.AspNetCore.RequestDecompression;
+
+/// <inheritdoc />
+public class RequestDecompressionProvider : IRequestDecompressionProvider
```

Comment:
I've updated this. It doesn't seem like this needs to be public, although we do have the `ResponseCompressionProvider` marked as public.

Now that this is internal, how should the `IRequestDecompressionProvider` interface be handled? I'd assume we don't want this interface to be part of the public API if it only relates to an internal implementation, correct?

Reply:
> Now that this is internal, how should the IRequestDecompressionProvider interface be handled? I'd assume we don't want this interface to be part of the public API if it only relates to an internal implementation, correct?

I think the implementation should be `internal` but the interface should be public in case a user wants to implement its own custom logic.